happytransformer/happy_generation.py

```diff
@@ -0,0 +1,163 @@
+"""
+Contains the HappyGeneration class
+"""
+from dataclasses import dataclass
+from transformers import AutoModelForCausalLM
+from happytransformer.happy_transformer import HappyTransformer
+from happytransformer.toc.trainer import TOCTrainer
+from happytransformer.adaptors import get_adaptor
+
+DEFAULT_SETTINGS = {
+    "do_sample": False,
+    "early_stopping": False,
+    "num_beams": 1,
+    "temperature": 0.65,
+    "top_k": 50,
+    "top_p": 1.0,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
+}
+
+default_greedy_settings = {
+    "do_sample": False,
+    "early_stopping": False,
+    "repetition_penalty": 1,
+    "length_penalty": 1,
+    "no_repeat_ngram_size": 2,
+    'bad_words_ids': None,
```

Comment:
Are these settings identical between methods by coincidence or are they inheriting the base settings?

Reply:
They're  inheriting the base settings. We can remove them, but I added them so that that when people copy the settings they see parameters they can play around with

Reply:
I changed the logic for the settings in 79fca23.

Reply:
The new changes allows users to use any of the parameters available for Hugging Face's generate() method.